web_client/dialogs/editRegionOfInterest.js

```diff
@@ -0,0 +1,207 @@
+        var scaleHeight = Math.round(factor * this.areaElement.height);
+        return { 'width': scaleWidth, 'height': scaleHeight };
+    },
+
+    /**
+     * Get the number of pixel in the region of interest
+     */
+    getNumberPixels() {
+        var bounds = this.scaleBounds();
+        var Npixel = bounds.width * bounds.height;
+        return Npixel;
+    },
+
+    /**
+     * Get the size of the file before download it for an image in 24b/px (result in Bytes)
+     */
+    getFileSize() {
+        var fileSize = (this.getNumberPixels() * 3 + this._sizeCte) * this._compressionRatio;
+        return fileSize;
+    },
+
+    /**
+     * Get the size of the file in the appropriate unity (Bytes, MB, GB...)
+     */
+    getConvertFileSize() {
+        var Nbytes = this.getFileSize();
+        var convertedSize = formatSize(Nbytes);
+        if (Nbytes >= Math.pow(2, 30)) {
+            this.downloadDisable(true);
+        } else {
+            this.downloadDisable(false);
+        }
+        return convertedSize;
+    },
+
+    /**
+     * Disable the Download button if SizeFile > 1GB
+     */
+    downloadDisable(bool) {
+        var element = $('#msgDisable').attr('id');
+        if (bool === true) {
+            $('#download-submit').attr('disabled', 'disabled');
+            if (typeof element === typeof undefined) {
+                var msgDisable = $('<span></span>').text('Size > 1GB : Impossible Download ');
+                msgDisable.attr('id', 'msgDisable');
+                msgDisable.css({'color': 'red', 'margin-right': '120px'});
+                $('#download-area-link').before(msgDisable);
+            }
+        } else if (bool === false) {
+            $('#download-submit').removeAttr('disabled');
+            if (typeof element !== typeof undefined) {
+                $('#msgDisable').remove();
+            }
+        } else {
+            console.log('Error in \'downloadDisable\' function');
+        }
+    },
+
+    /**
+     * Set the size of the file, bounds, format...
+     * And download the image
+     */
+    updateform(evt) {
+        // Find the good compresion ration there are random now
+        var selectedOption = $('#download-image-format option:selected').text();
+        switch (selectedOption) {
+            case 'JPEG':     //     JPEG
+                this._format = 'JPEG';
+                this._compressionRatio = 0.35;
+                break;
+            case 'PNG':     //  PNG
+                this._format = 'PNG';
+                this._compressionRatio = 0.7;
+                break;
+            case 'TIFF':     // TIFF
+                this._format = 'TIFF';
+                this._compressionRatio = 0.8;
+                break;
+            default:     // JPEG is the default format
+                this._compressionRatio = 0.35;
+        }
+        this._magnification = parseFloat($('#h-element-mag').val());
+        var bounds = this.scaleBounds();
```

Comment:
`var` definitions are always [hoisted](https://www.w3schools.com/js/js_hoisting.asp) to the top of the function.  It would be better to define the variables at the top or change all declarations to `let`/`const`.